dlib/array2d/array2d_kernel.h

```diff
@@ -489,8 +490,14 @@ namespace dlib
         {
             if (nr_ > 0)
             {
-                data = pool.allocate_array(nr_*nc_);
-                last = data + nr_*nc_ - 1;
+                const auto prod = dlib::detail::safe_multiply(nr_,nc_);
+#ifdef FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION
+                if(prod>200*200) {
```

Comment:
It's very reasonable to allocate arrays larger than 200x200.

Reply:
Yes, I agree! This is only during fuzzing, to speed it up. That's why it's inside the ifdef check.